hack/deployer/runner/ocp.go

```diff
@@ -462,3 +463,45 @@ func (d *OCPDriver) baseDomain() string {
 	}
 	return baseDomain
 }
+
+func (d *OCPDriver) Cleanup(prefix string, olderThan time.Duration) ([]string, error) {
+	if err := d.authToGCP(); err != nil {
+		return nil, err
+	}
+	daysAgo := time.Now().Add(-olderThan)
+
+	params := d.bucketParams()
+	params["Date"] = daysAgo.Format(time.RFC3339)
+	params["E2EClusterNamePrefix"] = prefix
+	params["Region"] = d.plan.Ocp.Region
+
+	if d.plan.Ocp.GCloudProject == "" {
+		gCloudProject, err := vault.Get(d.vaultClient, OCPVaultPath, GKEProjectVaultFieldName)
+		if err != nil {
+			return nil, err
+		}
+		d.plan.Ocp.GCloudProject = gCloudProject
+	}
+	params["GCloudProject"] = d.plan.Ocp.GCloudProject
+
+	zonesCmd := `gcloud compute zones list --verbosity error --filter='region:https://www.googleapis.com/compute/v1/projects/{{.GCloudProject}}/regions/{{.Region}}' --format="value(selfLink.name())"`
+	zones, err := exec.NewCommand(zonesCmd).AsTemplate(params).WithoutStreaming().OutputList()
+	if err != nil {
+		return nil, err
+	}
+	params["Zones"] = strings.Join(zones, ",")
+
+	cmd := `gcloud compute instances list --verbosity error --zones={{.Zones}} --filter="name~'^{{.E2EClusterNamePrefix}}-ocp.*' AND status=RUNNING" --format=json | jq -r --arg d "{{.Date}}" 'map(select(.creationTimestamp | . <= $d))|.[].name' | grep -o '{{.E2EClusterNamePrefix}}-ocp-[a-z]*-[0-9]*' | sort | uniq`
+	clustersToDelete, err := exec.NewCommand(cmd).AsTemplate(params).WithoutStreaming().OutputList()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cluster := range clustersToDelete {
+		d.plan.ClusterName = cluster
+		if err = d.delete(); err != nil {
```

Comment:
This is not going to work. You have to run through the setup steps as in `Execute` including `ensureClientImage`